db/fetcher/fetcher.go

```diff
@@ -48,14 +59,21 @@ type DocumentFetcher struct {
 	order        []dsq.Order
 	curSpanIndex int
 
-	schemaFields map[uint32]client.FieldDescription
-	fields       []*client.FieldDescription
+	filter  *mapper.Filter
+	mapping *core.DocumentMapping
+
+	filterFields map[uint32]client.FieldDescription
+	selectFields map[uint32]client.FieldDescription
+	// allfields       map[uint32]*client.FieldDescription
+
+	// static bitsets
+	filterSet *bitset.BitSet
```

Comment:
todo: The usage of this property is not self explanatory and needs code-comments, I am very tired atm, but it should still be easier to understand than it currently is for me :)